Fakebook.Profile/Fakebook.Profile.DataAccess/ProfileRepository.cs

```diff
@@ -69,75 +69,77 @@ private static DomainProfile ToDomainProfile(EntityProfile profile)
         /// Default model mapping from domain to entity.
         /// </summary>
         /// <param name="profile">domain model object used</param>
-        /// <exception type="ArgumentNullException">If the profile, the profile's email, or the profile's names are null,
+        /// <exception type="ArgumentException">If the profile, the profile's email, or the profile's names are null,
         /// this will be thrown </exception>
         /// <returns>A representation of the profile as a DB Entity.</returns>
-        private Task<EntityProfile> ToEntityProfile(DomainProfile profile)
+        private async Task<EntityProfile> ToEntityProfileAsync(DomainProfile profile)
         {
             if (profile == null || profile.Email == null || profile.FirstName == null || profile.LastName == null)
             {
-                throw new ArgumentNullException("Must have a domain profile, with an email and first and last name.");
+                throw new ArgumentException("Profile is missing required fields.", nameof(profile));
             }
 
-            return ToEntityProfileAsync(profile);
-        }
+            var entityProfile = await ConvertAsync(profile);
```

Comment:
i'm unsure if you're doing it wrong or if sonarqube just isn't smart enough to understand local functions yet, but i'm pretty sure you should make the method non-async, and just directly return the Task without awaiting it. in local function leave as-is.

it has some minor implications, but you don't actually have to await if it's the very last call in the method. whoever calls it will get that same Task object and await it for you.

Reply:
Removed async from ToEntityProfileAsync and removed await from ConvertAsync.